### ISSUE TITLE
Mac OS X Lion Windows Display

### ISSUE THREAD

**Issue by racket-bug-submit** [samth/test-bugs#139]
###### Originally submitted by Anh Luong luong@eng.utah.edu on: Wed Oct 05 10:56:01 -0400 2011

Windows when open up does not bring up to the front. I have to click on bring frame to the front to get the windows to display. I ussually just cancel out of that when windows show up. Once they show up, the red button to close out of windows doesn't close the windows and quit dr racket won't close all the windows and quit. It normally just hang and have to kill process manually. After running the program twice, it use up a tremedous amount of memory and slow down significantly.
##### Steps to Reproduce:

Start up DrRacket in Mac OS X Lion. Current version 10.7.1. Run a 300 lines program for more than 2 times.
##### Release:

```
5.1.3
```
##### Environment:

```
macosx "Darwin Shinigami 11.1.0 Darwin Kernel Version 11.1.0: Tue Jul 26 16:07:11 PDT
 2011; root:xnu-1699.22.81~1/RELEASE_X86_64 x86_64" (x86_64-macosx/3m) (get-display-depth)
 = 32
Human Language: english
(current-memory-use) 168035352

Collections:
("/Users/anhluong/Library/Racket/5.1.3/collects"
 ("info-domain" "uu-cs5510"))
("/Applications/Racket v5.1.3/collects"
 ("2htdp" "algol60" "at-exp" "browser" "combinator-parser" "compiler" "config" "data"
 "datalog" "defaults" "deinprogramm" "drracket" "drscheme" "dynext" "embedded-gui" "eopl"
 "errortrace" "ffi" "file" "framework" "frtime" "games" "graphics" "gui-debugger" "help"
 "hierlist" "htdp" "html" "icons" "info-domain" "lang" "launcher" "lazy" "macro-debugger"
 "make" "mred" "mrlib" "mysterx" "mzcom" "mzlib" "mzscheme" "net" "openssl" "parser-tools"
 "picturing-programs" "plai" "planet" "plot" "preprocessor" "profile" "r5rs" "r6rs"
 "racket" "racklog" "rackunit" "raco" "reader" "readline" "redex" "rnrs" "s-exp" "scheme"
 "schemeunit" "scribble" "scribblings" "scriblib" "setup" "sgl" "slatex" "slideshow"
 "srfi" "stepper" "string-constants" "swindle" "syntax" "syntax-color" "teachpack"
 "test-box-recovery" "test-engine" "tex2page" "texpict" "trace" "typed" "typed-scheme"
 "unstable" "version" "web-server" "wxme" "xml"))

Computer Language: (("Determine language from source") (#(#t print mixed-fraction-e #f #t
 debug) (default) #() "#lang racket\n" #t #t))
```
###### This bug was converted from Gnats bug [12257](http://bugs.racket-lang.org/old/12257).

[anon-submit luong AT eng.utah.edu]

**Comment by racket-bug-submit** [comment]
###### On Wed, 5 Oct 2011 09:14:03 -0600, mflatt@cs.utah.edu wrote:

``` racket
Hi Anh --- Are you running the 64-bit version of the 32-bit version?
```

**Comment by racket-bug-submit** [comment]
###### On Fri, 07 Oct 2011 10:07:12 -0400, mflatt@racket-lang.org assigned this bug to mflatt

A commit by mflatt@racket-lang.org was marked as relevant
  http://git.racket-lang.org/plt/commit/60101618db

**Comment by racket-bug-submit** [comment]
###### On Sun, 27 Nov 2011 11:32:09 -0500, mflatt wrote:

fixed in v5.2
